src/main/app/pay/payClient.ts

```diff
@@ -1,54 +1,103 @@
 import * as config from 'config'
-
-import { PaymentRequest } from 'app/pay/paymentRequest'
-import { PaymentResponse } from 'app/pay/paymentResponse'
+import { PaymentRetrieveResponse } from 'app/pay/paymentRetrieveResponse'
 import { Payment } from 'app/pay/payment'
 import { request } from 'client/request'
 import { User } from 'app/idam/user'
 import { ServiceAuthToken } from 'app/idam/serviceAuthToken'
-
+import * as HttpStatus from 'http-status-codes'
 import * as uuid from 'uuid/v4'
+import { Fee } from 'app/pay/fees'
+import { plainToClass } from 'class-transformer'
+
+const baseURL = `${config.get('pay.url')}/card-payments`
 
-const payUrl = config.get('pay.url')
+const serviceName = config.get<string>('pay.service-name')
+const currency = config.get<string>('pay.currency')
+const siteId = config.get<string>('pay.site-id')
+const description = config.get<string>('pay.description')
 
 export class PayClient {
   constructor (public serviceAuthToken: ServiceAuthToken) {
     this.serviceAuthToken = serviceAuthToken
   }
 
   /**
-   * Creates a payment request for CC-Pay
+   * Creates a payment within Reform Payment Hub
    *
-   * @param user a user
-   * @param feeCode the fee code
-   * @param amount the amount in pounds
-   * @param returnURL the url the user should be redirected to
-   * @returns {Promise.URL}
+   * @param user - user who make a call
+   * @param fees - fees array used to calculate total fee amount
+   * @param amount - total fee amount
+   * @param returnURL - the url the user should be redirected to
+   * @returns response with payment status and link to card payment page
    */
-  create (user: User, feeCode: string, amount: number, returnURL: string): Promise<PaymentResponse> {
-    const paymentReference = `CMC1$$$${uuid()}$$$AA00$$$${feeCode}`
+  async create (user: User, fees: Fee[], amount: number, returnURL: string): Promise<Payment> {
+    if (!user) {
+      throw new Error('User is required')
+    }
+    if (!fees) {
+      throw new Error('Fees array is required')
+    }
+    if (!amount) {
+      throw new Error('Total fee amount is required')
+    }
+    if (!returnURL) {
```

Comment:
This code fragment to validate if parameter is not blank is used multiple times across this PR. What do you think about extracting a function like e.g. `notBlank(value: any, requiredMessage:string)`?

Reply:
Good suggestion, I created `commons/preconditions.ts` with few functions.